### DISCUSSION TITLE
Help Regarding OpenerService

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#10296] [General] [answered]
Hi,
I'm having trouble understanding OpenerService like how does is work and it navigates to elements. Is there a simple working example to understand about it ? Also, if possible kindly provide documentation links too. 

Thanks

**Comment by vince-fugnitto** [comment, marked as answer]
@maheshrajrp thank you for the discussion, do you have a specific question about `OpenHandler` you do not understand?

For reference you can read the documentation of [opener-service.ts](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/opener-service.ts) and for an example you can take a look at any present in the codebase (searching for references) (ex: [preview-contribution.ts](https://github.com/eclipse-theia/theia/blob/54b6ced3178f77ea449cdefb502b9cdcfec80190/packages/preview/src/browser/preview-contribution.ts#L54) which handles opening specific files such as '.svg' or '.markdown' as previews).

**Reply by maheshrajrp** [reply]
I was hoping for an answer like what problem does Opener Service solve just for basic understanding.
